### ISSUE TITLE
Cannot do second partial invoice after first partial invoice

### ISSUE THREAD

**Issue by ghost** [magento/magento2#23946]
### Preconditions (*)

1. Magento 2.2.8
2. Braintree setup to authorize only

### Steps to reproduce (*)

1. Add 2+ products to cart
2. Order products as guest using Credit Card
3. In order admin, invoice for 1 item, This should go through fine
4. Invoice again for 1 more item. Error will occur

### Expected result (*)

1. Second invoice should go through same as the first

### Actual result (*)

1. Second invoice does not go through

It looks like the capture strategy that is selected is `VAULT_CAPTURE` but this cannot work for guests, as their card details are not stored and therefore `$paymentToken = $extensionAttributes->getVaultPaymentToken();` returns `null`

I believe the capture strategy should be `CAPTURE` but if this is used, the checks done in `CaptureStrategyCommand` class won't pick this strategy.

Looks like the logic 

```
if (!$existsCapture && !$this->isExpiredAuthorization($payment, $paymentDO->getOrder())) {
            return self::CAPTURE;
        }
```

is wrong, but if i force this strategy, further errors are encountered, with a response saying the transaction status in Braintree needs to be `authorized` but it'll likely be on `submitted_for_settlement` because the initial invoice was raised.

**Comment by engcom-Bravo** [comment]
Hello @paul-gene. Could You please try to reproduce Your issue not on 2.2.8 but on the fresh, vanilla 2.3-develop instance as You are advised to do in one of the [above](https://github.com/magento/magento2/issues/23946#issuecomment-516390971) bot's comments. Thank You.

**Comment by ghost** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @paul-gene. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @paul-gene, here is your [Magento instance](https://i-23946-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-23946-2-3-develop.instances.magento-community.engineering/admin](https://i-23946-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by ghost** [comment]
@magento give me 2.2-develop instance

**Comment by magento-engcom-team** [comment]
Hi @paul-gene. Thank you for your request. I'm working on Magento 2.2-develop instance for you

**Comment by ghost** [comment]
OK, so it looks like this cannot be replicated on a fresh install, due to another bug in Braintree.

In core Braintree, card details are stored regardless of whether the config is set not to store cards.

https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Braintree/Gateway/Request/VaultDataBuilder.php#L33

Because this is always `true`, everyones cards are stored in the DB, regardless of the config settings, and therefore the capture strategy of `VAULT_CAPTURE` works.

If the VaultDataBuilder class is "fixed" to stop cards being stored all the time, then the `VAULT_CAPTURE` strategy will not work for a 2nd invoice, as no token is in the DB.

**Comment by engcom-Bravo** [comment]
Hello again @paul-gene. As we can conclude taking into consideration [Your comment to Your another issue](https://github.com/magento/magento2/issues/23949#issuecomment-516783223) this issue must also have been fixed by #19767 in 2.3-develop version. So due to our workflow we are closing it now. Please feel free to reopen if deem necessary.
